bin/improver-threshold

```diff
@@ -0,0 +1,105 @@
+"""Script to apply thresholding to a cube."""
+
+import argparse
+
+import iris
+import cf_units
+
+from improver.threshold import BasicThreshold
+from improver.utilities.cube_manipulation import concatenate_cubes
+from improver.spotdata.extract_data import ExtractData
+
+
+def main():
+    """Load in arguments and get going."""
+    parser = argparse.ArgumentParser(
+        description="Calculate the threshold truth value of cube data "
+        "relative to the provided threshold value. A fuzzy factor may be "
+        "provided to capture data that is within this factor of the "
+        "threshold.")
+    parser.add_argument("input_filepath", metavar="INPUT_FILE",
+                        help="A path to an input NetCDF file to be processed")
+    parser.add_argument("output_filepath", metavar="OUTPUT_FILE",
+                        help="The output path for the processed NetCDF")
+    parser.add_argument("threshold_values", metavar="THRESHOLD_VALUES",
+                        nargs="+", type=float,
+                        help="Threshold value or values about which to "
+                        "calculate the truth values; e.g. 270 300")
+    parser.add_argument("--threshold_units", metavar="THRESHOLD_UNTIS",
+                        default=None, type=str,
+                        help="Units of the threshold values. If not provided "
+                        "the units are assumed to be the same as those of the "
+                        "input cube. Specifying the units here will allow a "
+                        "suitable conversion to match the cube units if "
+                        "possible.")
+    parser.add_argument("--below_threshold", metavar="BELOW_THRESHOLD",
+                        default=False, type=bool,
+                        help="By default truth values of 1 are returned for "
+                        "data ABOVE the threshold value(s). Setting this "
+                        "to TRUE changes this behaviour to return 1 for data "
+                        "below the threshold values.")
+    parser.add_argument("--fuzzy_factor", metavar="FUZZY_FACTOR",
+                        default=None, type=float,
+                        help="A decimal fraction defining the factor about "
+                        "the threshold value(s) which should be treated as "
+                        "fuzzy. Data which fail a test against the hard "
+                        "threshold value may return a fractional truth value "
+                        "if they fall within this fuzzy factor region.")
+
+    args = parser.parse_args()
+    cube = iris.load_cube(args.input_filepath)
+
+    # Allow for threshold value unit conversion.
+    if args.threshold_units is not None:
+        threshold_unit = cf_units.Unit(args.threshold_units)
+        args.threshold_values = [threshold_unit.convert(threshold, cube.units)
+                                 for threshold in args.threshold_values]
+        print args.threshold_values
```

Comment:
Need to get rid of this print statement.

Reply:
:white_check_mark: